Add UID to ROI objects

hackermd opened the pull request:
This PR adds a `uid` property to `ROI` objects, which will facilitate tracking of regions and measurement groups.

hackermd commented:
While refactoring the code, I realized that the required [Referenced Frame of Reference UID](http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.18.9.html#para_14610131-0a52-4367-8ebe-36688ff8305d) attribute was missing on `Scoord3D` objects. I added this in 2181fc1. This required changes to the constructor of `Scoord3D` and derived classes. They now all accept and `options` object.

swederik commented on src/api.js:
```diff
@@ -828,6 +854,7 @@ class VLWholeSlideMicroscopyImageViewer {
     const geometry = _scoord3d2Geometry(item.scoord3d, this._pyramid);
     const feature = new Feature(geometry);
     feature.setProperties(item.properties, true);
+    feature.setId(item.uid);
```
Maybe we should generate the `id` here if `item.uid` is undefined?

hackermd replied:
Addressed with a88b5879b0d3627c5430ca0bd348d32577b628fe

biharck commented on src/api.js:
```diff
@@ -40,12 +41,16 @@ import {
-    return new Point(coordinates);
+    return new Point({
+      coordinates,
+      referencedFrameOfReferenceUID: pyramid[pyramid.length-1].frameOfReferenceUID
```
I would move `pyramid[pyramid.length-1].frameOfReferenceUID`  to after line 46 and extract as a constant since it is the same for all geometries.

`const referencedFrameOfReferenceUID = pyramid[pyramid.length-1].frameOfReferenceUID `

```
return new Polygon({
      coordinates,
      referencedFrameOfReferenceUID
    });

...
return new Polyline({
      coordinates,
      referencedFrameOfReferenceUID
    });
...
etc
```

hackermd replied:
Addressed with 707825a858700025c7520e7d2678099f3edc9d35

pieper commented on src/metadata.js:
```diff
@@ -40,6 +40,7 @@ function formatImageMetadata(metadata) {
   }
 
   const imageOrientationSlide = metadata['00480102']['Value'];
+  const frameOfReferenceUID = metadata['00200052']['Value'][0];
```
This code is consistent with the rest of the package, but it made me with we didn't have hex everywhere.  I added an issue for this.

https://github.com/dcmjs-org/dicom-microscopy-viewer/issues/27

hackermd replied:
I generally agree with you on the use of keywords over tags. Let's discuss in #27 and address this with another PR, ok?

hackermd commented:
I stumbled upon a couple of issues related to spatial coordinates:

1. There is no CIRCLE 3D spatial coordinate, but an ELLIPSOID.
2. Coordinate values should be positive integers. I fixed the conversion between OpenLayers map coordinate system (inverted y-axis) and the DICOM slide coordinate system.

In addition, I replaced the UUID with a DICOM UID with root `"2.25"`. To this end, I used the [DicomMetaDictionary.uid()](https://github.com/dcmjs-org/dcmjs/blob/3e6e416233e7752068cc79f8e93212afacb07b9b/src/DicomMetaDictionary.js#L220-L226) implementation in *dcmjs*. However, this does not create a valid DICOM UID (see https://github.com/dcmjs-org/dcmjs/issues/61).

@swederik @biharck @pieper could you please have a second look?
